checks/doks/dobs_pod_owner_test.go

```diff
@@ -316,6 +437,20 @@ func deployment(objs *kube.Objects) *kube.Objects {
 	return objs
 }
 
+// Users request dynamically provisioned storage by including a storage class in their PersistentVolumeClaim.
+// Before Kubernetes v1.6, this was done via the volume.beta.kubernetes.io/storage-class annotation. However, this annotation is deprecated since v1.9. 
+// https://kubernetes.io/docs/concepts/storage/dynamic-provisioning/#:~:text=Users%20request%20dynamically,the%20PersistentVolumeClaim%20object.
+func provideStorageClassInAnnotation(objs *kube.Objects) *kube.Objects {
+	if objs.PersistentVolumeClaims.Items[0].Spec.StorageClassName != nil {
+		if objs.PersistentVolumeClaims.Items[0].Annotations == nil {
+			objs.PersistentVolumeClaims.Items[0].Annotations = make(map[string]string)
+		}
+		objs.PersistentVolumeClaims.Items[0].Annotations["volume.beta.kubernetes.io/storage-class"] = *objs.PersistentVolumeClaims.Items[0].Spec.StorageClassName
```

Comment:
nit: I think here you could do the creation of the map and setting a value inside that in one step if I'm not mistaken?

Reply:
AFAIU, the given current is to not clear any existing annotations that might already be on the given object.

Reply:
I did it this way because, if the PVC already has annotations I don't want to override them with a new map.

Reply:
Ah yes you want it set in both the cases not just one case where annotations was nil makes sense